### PR TITLE
Resolve p3a related issue for VPN

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/24797

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Please follow the main issue test plan

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/vpn/utils/BraveVpnProfileUtils.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ public boolean isVPNRunning(Context context) {
     }
 
     public void startVpn(Context context) {
-        BraveVpnNativeWorker.getInstance().reportForegroundP3A();
+        // BraveVpnNativeWorker.getInstance().reportForegroundP3A();
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/vpn/utils/BraveVpnProfileUtils.java:66]: @DJAndries found an issue here. it's calling the native function and when we try to change the server, it crashes. so for now i have commented out the code.

File: android/java/org/chromium/chrome/browser/vpn/wireguard/WireguardServiceImpl.java [hunk 2]
```diff
@@ -152,8 +153,8 @@ private void recordSessionTimes() {
     private void updateRecordSessionTimesTimer() {
         // This timer will be active for the duration of the connection.
         // It will become inactive when the connection is terminated/service destroyed.
-        mVpnStatisticsTimer = new Timer();
-        mVpnStatisticsTimer.schedule(new TimerTask() {
+        mVpnRecordStatisticsTimer = new Timer();
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/vpn/wireguard/WireguardServiceImpl.java:156]: We need to use different object for timer as using the same object was affecting the notification update for the VPN.

Comment (DJAndries) [reply android/java/org/chromium/chrome/browser/vpn/wireguard/WireguardServiceImpl.java:156]: My mistake, I meant to use `mRecordDaysUsedTimer`. I will remove `mRecordDaysUsedTimer` in a new commit in this branch.

**Review by DJAndries — APPROVED** [review]
Will resolve in separate PR

**Comment by kjozwiak** [comment]
Reproduced the original issue using `Samsung S10+` running `Android 12` using `1.44.60 Chromium: 104.0.5112.102` as per the following:

https://user-images.githubusercontent.com/2602313/187098873-134aed3a-8f20-4d94-a1bd-f590e3bcbc6b.mp4

Verification **PASSED** on `Samsung S10+` running `Android 12` using the following build(s):

```
Brave | 1.45.7 Chromium: 105.0.5195.58 (Official Build) canary (32-bit)
--- | ---
Revision | 1907e1693887af4463221a10247360b7a056db49-refs/branch-heads/5195_47@{#3}
OS | Android 12; Build/SP1A.210812.016
```

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/24797#issue-1343924066, ensured that switching between different VPN servers didn't crash the browser like it did with `1.44.60 Chromium: 104.0.5112.102` as per the above.

**`Example of issue fixed:`**

https://user-images.githubusercontent.com/2602313/187099303-ec7acdfe-0b2f-45ec-9815-f0ba95cf0ed8.mp4

**Pull request merged**